From 7855094c8f211bcc11771525397420d52cbd0f65 Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:29 -0500
Subject: [wireless] Revert: ath9k: reduce ANI firstep range for older chips

Message-id: <1418207389-7899-16-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101585
O-Subject: [RHEL7.1 PATCH 15/35] Revert "ath9k_hw: reduce ANI firstep range for older chips"
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

From: Felix Fietkau <nbd@openwrt.org>

BZ#1155538

commit 171cdab8c78bb169d9693d587e1d02d2dd5a0274 upstream.

This reverts commit 09efc56345be4146ab9fc87a55c837ed5d6ea1ab

I've received reports that this change is decreasing throughput in some
rare conditions on an AR9280 based device

Signed-off-by: Felix Fietkau <nbd@openwrt.org>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Compiled only.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/ath/ath9k/ar5008_phy.c b/drivers/net/wireless/ath/ath9k/ar5008_phy.c
index 00fb8ba..3b3e910 100644
--- a/drivers/net/wireless/ath/ath9k/ar5008_phy.c
+++ b/drivers/net/wireless/ath/ath9k/ar5008_phy.c
@@ -1004,9 +1004,11 @@ static bool ar5008_hw_ani_control_new(struct ath_hw *ah,
  case ATH9K_ANI_FIRSTEP_LEVEL:{
   u32 level = param;
 
-  value = level;
+  value = level * 2;
   REG_RMW_FIELD(ah, AR_PHY_FIND_SIG,
          AR_PHY_FIND_SIG_FIRSTEP, value);
+  REG_RMW_FIELD(ah, AR_PHY_FIND_SIG_LOW,
+         AR_PHY_FIND_SIG_FIRSTEP_LOW, value);
 
   if (level != aniState->firstepLevel) {
    ath_dbg(common, ANI,
-- 
1.7.1